provisioning_templates/snippet/set_hostname.erb

```diff
@@ -0,0 +1,20 @@
+<%#
+name: set_hostname
+snippet: true
+model: ProvisioningTemplate
+kind: snippet
+-%>
+if [ `command -v hostnamectl` ]; then
+  hostnamectl set-hostname <%= @hostname %>
+else
+  hostname <%= @hostname %>
+  # persist the change to survive  reboot
+  echo "<%= @hostname -%>" > /etc/hostname
+fi
+
+if grep '^127.0.0.1\s*<%= @hostname -%>$' /etc/hosts; then
+  echo "skipping /etc/hosts update, record is already present"
+else
+  echo "adding record to the first line of /etc/hosts"
+  sed -i '1s/^/127.0.0.1   <%= @hostname -%>\n/' /etc/hosts
```

Comment:
I don't know... at least Debian/Ubuntu sets `127.0.1.1 fqdn hostname` per default. I'd rather go for 127.0.0.1 for localhost and the primary IP for the fqdn and hostname...